src/generate/generic.rs

```diff
@@ -10,6 +10,27 @@ pub fn render() -> Result<Vec<Tokens>> {
     generic_items.push(quote! {
         use core::marker;
 
+        ///This trait shows that register has `read` method
+        ///
+        ///Registers marked with `Writable` can be also `modify`'ed
+        pub trait Readable {}
+
+        ///This trait shows that register has `write`, `write_with_zero` and `reset` method
+        ///
+        ///Registers marked with `Readable` can be also `modify`'ed
+        pub trait Writable {}
+
+        ///Reset value of the register
+        ///
+        ///This value is initial value for `write` method.
+        ///It can be also directly writed to register by `reset` method.
+        pub trait ResetValue {
+            ///Register size
+            type Size;
+            ///Reset value of the register
+            fn reset_value() -> Self::Size;
```

Comment:
This associated type name is debatable, but I'm not going to force my point of view here.

Reply:
I've changed Size on Type